lib/ClangImporter/ImportDecl.cpp

```diff
@@ -8865,21 +8869,48 @@ class SwiftifyInfoPrinter {
   }
 
 private:
+  void printParamOrReturn(ssize_t pointerIndex) {
+    if (pointerIndex == -2)
+      out << ".self";
+    else if (pointerIndex == -1)
+      out << ".return";
+    else
+      out << ".param(" << pointerIndex + 1 << ")";
+  }
+
+protected:
   void printSeparator() {
     if (!firstParam) {
       out << ", ";
     } else {
       firstParam = false;
     }
   }
+};
 
-  void printParamOrReturn(ssize_t pointerIndex) {
-    if (pointerIndex == -2)
-      out << ".self";
-    else if (pointerIndex == -1)
-      out << ".return";
-    else
-      out << ".param(" << pointerIndex + 1 << ")";
+class SwiftifyProtocolInfoPrinter : SwiftifyInfoPrinter {
+public:
+  SwiftifyProtocolInfoPrinter(clang::ASTContext &ctx, llvm::raw_ostream &out)
+      : SwiftifyInfoPrinter(ctx, out) {}
+
+  bool printMethod(const FuncDecl *Method,
+                   bool (*importFunctionBoundsAttributes)(
+                       SwiftifyInfoPrinter &, const clang::ObjCMethodDecl *)) {
+    const auto *ClangDecl =
+        dyn_cast_or_null<clang::ObjCMethodDecl>(Method->getClangDecl());
+    if (!ClangDecl)
+      return false;
+
+    printSeparator();
+    out << ".method(name: \"" << Method->getName().getBaseIdentifier()
```

Comment:
Will this work for overloaded methods?

Reply:
Hmm that's a really good point actually... I'm guessing macros don't have access to mangled names of declarations, otherwise comparing the mangled name instead could work.

Reply:
This now passes the entire method signature to disambiguate overloaded methods